src/workers/_inputs.js

```diff
@@ -37,6 +37,19 @@ function dummyGenes(numberOfRowss) {
     return { "id": genes };
 }
 
+function readCSVFromBuffer(content, compressed) {
+    if (compressed) {
+        content = pako.ungzip(content);
+    }
+
+    const dec = new TextDecoder();
+    let decoded = dec.decode(content);
+    const tsv = d3.dsvFormat("\t");
```

Comment:
Perhaps say `readDSVFromBuffer` and allow callers to supply a delimiter based on the file extension.